clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts

```diff
@@ -36,10 +37,12 @@ export class IModelsErrorBaseImpl extends Error {
 
 export class IModelsErrorImpl extends IModelsErrorBaseImpl implements IModelsError {
   public details?: IModelsErrorDetail[];
+  public statusCode?: number;
 
-  constructor(params: { code: IModelsErrorCode, message: string, details?: IModelsErrorDetail[] }) {
+  constructor(params: { code: IModelsErrorCode, message: string, details?: IModelsErrorDetail[], statusCode?: number }) {
```

Comment:
could we make `statusCode` and `details` mandatory params? Having `statusCode` as an optional argument makes it easier for us to leave a bug and not pass it to constructor.

Reply:
Mandatory as passing 'undefined' explicitly?

Reply:
yes

Reply:
updated